Tests/Shared.Specs/TypeAssertionSpecs.cs

```diff
@@ -1445,6 +1446,50 @@ public void When_injecting_a_null_predicate_into_TypeSelector_NotBeDecoratedWith
 
         #endregion
 
+        #region BeInNamespace
+
+        [Fact]
+        public void When_asserting_a_selection_of_types_in_a_namespace_is_in_that_namespace_it_succeeds()
+        {
+            // Arrange
+            var types = new TypeSelector(new[]
+            {
+                typeof(ClassInDummyNamespace)
+            });
+
+            // Act
+            Action act = () =>
+                types.Should().BeInNamespace(nameof(DummyNamespace));
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_a_selection_of_types_in_a_namespace_is_in_another_namespace_it_fails()
+        {
+            // Arrange
+            var types = new TypeSelector(new[]
+            {
+                typeof(ClassInDummyNamespace),
+                typeof(ClassNotInDummyNamespace),
+                typeof(OtherClassNotInDummyNamespace)
+            });
+
+            // Act
+            Action act = () =>
+                types.Should().BeInNamespace(nameof(DummyNamespace), "because reasons");
```

Comment:
One minor thing, please also include a single `becauseArgs`.
We usually use something alike
`"because we want to test the error {0}", "message"`

Reply:
Will do.